Alert when an ingester replica does not have any tenant assigned (#3681)
grafana/mimir: docs/sources/mimir/operators-guide/mimir-runbooks/_index.md

- Look for details in the ingester logs

### MimirIngesterInstanceHasNoTenants

This alert fires when an ingester instance doesn't own any tenants and is therefore idling.

How it **works**:

- Ingesters join a hash ring that facilitates per-tenant request sharding across ingester replicas.

> - Distributors shard requests belonging to an individual tenant across a subset of ingester replicas. The number of replicas used per tenant is determined by the `-distributor.ingestion-tenant-shard-size` or the `ingestion_tenant_shard_size` limit.

Reviewer: ```suggestion
- Distributors shard requests that belong to an individual tenant across a subset of ingester replicas. The number of replicas used per tenant is determined by the `-distributor.ingestion-tenant-shard-size` or the `ingestion_tenant_shard_size` limit.
```

> - When the tenant shard size is lower than the number of ingester replicas, some ingesters may not receive requests for any tenants.

Reviewer: ```suggestion
- When the tenant shard size is lower than the number of ingester replicas, some ingesters might not receive requests for any tenants.
```

Reviewer: Rationale: Change spoken language to written language; disambiguate "may" (permission) from "might" (possibility).

- This is more likely to happen in Mimir clusters with a lower number of tenants.

How to **fix** it:

> There are three options:

Reviewer: ```suggestion
Choose one of three options:
```

> - Decrease the number of ingester replicas to match the highest number of shards per tenant or

Reviewer: ```suggestion
- Decrease the number of ingester replicas to match the highest number of shards per tenant.
```

> - Decrease the number of ingester replicas to match the highest number of shards per tenant.

Reviewer 2: Scaling down ingesters is a tricky procedure. It's not a "quick fix". https://grafana.com/docs/mimir/latest/operators-guide/run-production-environment/scaling-out/#scaling-down-ingesters

Perhaps we should suggest next two items first.

> - Increase the shard size of one or more tenants to match the number of ingester replicas or

Reviewer: ```suggestion
- Increase the shard size of one or more tenants to match the number of ingester replicas.
```

> - Set the shard size of one or more tenants to `0`; this will shard this tenant's requests across all ingesters.

Reviewer: ```suggestion
- Set the shard size of one or more tenants to `0`; this will shard a given tenant’s requests across all ingesters.
```
